sklego/linear_model.py

```diff
@@ -1128,15 +1131,16 @@ def __init__(
 
     def _get_objective(self, X, y, sample_weight):
         def quantile_loss(params):
-            return np.mean(
-                sample_weight * np.where(X @ params < y, self.quantile, 1 - self.quantile) * np.abs(y - X @ params)
+            return np.average(
+                np.where(X @ params < y, self.quantile, 1 - self.quantile) * np.abs(y - X @ params),
+                weights=sample_weight,
             ) + self._regularized_loss(params)
 
         def grad_quantile_loss(params):
             return (
                 -(sample_weight * np.where(X @ params < y, self.quantile, 1 - self.quantile) * np.sign(y - X @ params))
                 @ X
-                / X.shape[0]
+                / sample_weight.sum()
```

Comment:
Hey @Garve, sorry to drag you into this mess.

No need for you to read all the changes. TL;DR is that in this PR I added tests using [scikit-learn parametrize_with_checks](https://scikit-learn.org/dev/modules/generated/sklearn.utils.estimator_checks.parametrize_with_checks.html#sklearn.utils.estimator_checks.parametrize_with_checks)  to check compatibility.

As some of the tests were failing I took a closer look and adjusted the formulas to take into account `sample_weight`. However we still have tests failing for _just_ one of the solvers. I wonder if you have any clue or insight on why this is the case.

Reply:
If it's really just one solver I would just not sweat it. We should document it for sure but we can always patch that later.

Reply:
Apparently, for ubuntu and windows 'TNC' as well is failing, but only for `quantile=0.3` and `fit_intercept=False`, not in the other cases.
I am skipping that as well and improving the warning in the class description